pyproject.toml

```diff
@@ -73,3 +73,85 @@ testpaths = ["tests"]
 
 [tool.coverage.run]
 branch = true
+
+[tool.ruff]
+src = ["src"]
+exclude = ["docs/*", "analysis/*", "codebuild/*"]
```

Comment:
I went ahead and ran `format` on `analysis` and `codebuild` but I didn't think it would be worth it to actively maintain them style-wise